react/features/calendar-sync/functions.native.js

```diff
@@ -9,6 +13,39 @@ export * from './functions.any';
 
 const logger = require('jitsi-meet-logger').getLogger(__filename);
 
+/**
+ * Adds a Jitsi link to a calendar entry.
+ *
+ * @param {Object} state - The Redux state.
+ * @param {string} id - The ID of the calendar entry.
+ * @param {string} link - The link to add info with.
+ * @returns {Promise<*>}
+ */
+export function addLinkToCalendarEntry(
+        state: Object, id: string, link: string) {
+    return new Promise((resolve, reject) => {
+        getShareInfoText(state, link, true).then(shareInfoText => {
```

Comment:
If you `return getShareInfoText` you don't need the extra promise wrapper but you'd have to re-organize how you return the promises below.

Reply:
Good catch, thanks!

Reply:
No, sorry, I was too quick. Did you mean like this? Because if so, this is not valid promise handling. Or did I miss something?

```
export function addLinkToCalendarEntry(
        state: Object, id: string, link: string) {
    return getShareInfoText(state, link, true).then(shareInfoText => {
        RNCalendarEvents.findEventById(id).then(event => {
            const updateText = `${event.description}\n\n${shareInfoText}`;
            const updateObject = {
                id: event.id,
                ...Platform.select({
                    ios: {
                        notes: updateText
                    },
                    android: {
                        description: updateText
                    }
                })
            };

            return RNCalendarEvents.saveEvent(event.title, updateObject);
        });
    });
}
```

Reply:
If you choose it'd be something like
```
 return getShareInfoText(state, link, true)
    .then(shareInfoText => RNCalendarEvents.findEventById(id))
    .then(event => {
        const updateText = `${event.description}\n\n${shareInfoText}`;
        const updateObject = {
            id: event.id,
            ...Platform.select({
                ios: {
                    notes: updateText
                },
                android: {
                    description: updateText
                }
            })
        };

        return RNCalendarEvents.saveEvent(event.title, updateObject);
    });
```